src/Fixer/Comment/NoUselessCommentFixer.php

```diff
@@ -0,0 +1,115 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Fixer\Comment;
+
+use PhpCsFixer\AbstractFixer;
+use PhpCsFixer\FixerDefinition\CodeSample;
+use PhpCsFixer\FixerDefinition\FixerDefinition;
+use PhpCsFixer\Preg;
+use PhpCsFixer\Tokenizer\Token;
+use PhpCsFixer\Tokenizer\Tokens;
+
+/**
+ * @author Kuba Werłos <werlos@gmail.com>
+ */
+final class NoUselessCommentFixer extends AbstractFixer
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function getDefinition()
+    {
+        return new FixerDefinition(
+            'There must be no comment like "Class Foo".',
+            [
+                new CodeSample('<?php
+/**
+ * Class Foo
+ * Class to do something
+ */
+class Foo {}
+'),
+                new CodeSample('<?php
+class Foo {
+    /**
+     * Get bar
+     */
+    function getBar() {
+        return "bar";
+    }
+}
+'),
+            ]
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isCandidate(Tokens $tokens)
+    {
+        return $tokens->isAnyTokenKindsFound([T_COMMENT, T_DOC_COMMENT]);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getPriority()
+    {
+        // must be run before NoEmptyCommentFixer, NoEmptyPhpdocFixer, PhpdocTrimConsecutiveBlankLineSeparationFixer and PhpdocTrimFixer
+        return 6;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function applyFix(\SplFileInfo $file, Tokens $tokens)
+    {
+        foreach ($tokens as $index => $token) {
+            if (!$token->isGivenKind([T_COMMENT, T_DOC_COMMENT])) {
+                continue;
+            }
+
+            $nextIndex = $tokens->getTokenNotOfKindSibling(
+                $index,
+                1,
+                [[T_WHITESPACE], [T_COMMENT], [T_DOC_COMMENT], [T_ABSTRACT], [T_FINAL], [T_PUBLIC], [T_PROTECTED], [T_PRIVATE], [T_STATIC]]
+            );
+            if (null === $nextIndex) {
+                continue;
+            }
+
+            if ($tokens[$nextIndex]->isGivenKind([T_CLASS, T_INTERFACE, T_TRAIT])) {
+                $newContent = Preg::replace(
+                    '/((^|\R)\h*(#|\/*\**))\h*\b(Class|Interface|Trait)\h+[A-Za-z0-1\\\\_]+\.?(\h*\R\h*\*?|\h*$)/i',
+                    '$1',
+                    $token->getContent()
+                );
+            } elseif ($tokens[$nextIndex]->isGivenKind(T_FUNCTION)) {
+                $newContent = Preg::replace(
+                    '/((^|\R)\h*(#|\/*\**))\h*\b((Gets?|SetS?)\h+[A-Za-z0-1\\\\_]+|[A-Za-z0-1\\\\_]+\h+constructor)\.?(\h*\R\h*\*?|\h*$)/i',
```

Comment:
> Should it? (I think not)

Why not? It is useless.

Reply:
That's the question though, it really depends on what is stated. I'd say if it's not the same, don't remove it.

Reply:
I assumed that "set"/"get" + 1 word is useless - I cannot imagine example when it could be helpful. I'd like the "Gets A" to be removed if someone tries to add it for "setA" function.

Reply:
I'll leave the final call to the maintainers.

I'd like to enable this fixer in my projects as well, however the current state makes it too risky for me 😞